### PR TITLE
applications: nrf_desktop: Add documentation of output HID report

### PR DESCRIPTION
**Pull request by zycz**
Change adds documentation to changes introduced by: NCSDK-9909.

Jira: NCSDK-10267

Signed-off-by: Jan Zyczkowski <Jan.Zyczkowski@nordicsemi.no>

### CODE REVIEW

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -176,8 +176,41 @@ When configuring HEAP, make sure that the values for the following options match
...
+
+* HID mouse report,
+* HID keyboard report,
+* HID consumer control report,
+* HID system control report.
+
+Every of these reports uses predefined report format and is dedicated to provide given information.
+For example, the mouse motion is forwarded as HID mouse report.
+
+Given nRF Desktop device supports selected subset of the HID input reports.
```
Comment (pdunaj) [inline applications/nrf_desktop/README.rst:207]: s/nRF Desktop/the nRF Desktop
probably in many places

**Review by pdunaj — APPROVED** [review]
text is ok but there are some language related things to fix
leaving that to techwriters

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -176,8 +176,41 @@ When configuring HEAP, make sure that the values for the following options match
...
+HID data handling
+=================
+
+HID data can be exchanged using one of the following HID report types:
+
+* HID input report,
+* HID output report,
+* HID feature report.
+
+nRF Desktops uses all of these report types.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:188]: fixed

File: applications/nrf_desktop/README.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/README.rst:207]: fixed

**Comment by MarekPieta** [comment]
@greg-fer, could you review, please?

**Review by greg-fer — CHANGES_REQUESTED** [review]
The changes are so extensive I need a call with @MarekPieta to discuss some of them.

**Review by greg-fer — APPROVED** [review]

**Pull request merged**